### ISSUE TITLE
Can we lose `<title>` attributes when migrating children of controlaccess

### ISSUE THREAD

**Issue by rockivist** [SAA-SDT/EAD2002toEAD3#3]
Standard migration path for `<title>` linking attributes is to wrap that `<title>` inside `<ref>`.  However, ref isn't a valid child of controlaccess, so there's no obvious way to migrate the linking attribtues of title if a given title exist as as a child of controlaccess.  Can we live with this?

**Comment by rockivist** [comment]
Addressed by converting controlaccess/title to controlaccess/p/ref/title when controlaccess/title has linking attributes. A little over the top, but hopefully this is an edge case. Closing.

**Comment by MicheleCombs** [comment]
I’m not seeing why title can’t be directly converted, like all other controlaccess children?  It corresponds to a common MARC field (630), the same way subject is 650 and persname is 610.  We use it a lot since many of our collections pertain to authors, publishers, etc.  Is there a reason we have to lose it, or use this complicated solution instead of the simple one?

Michele

**Comment by rockivist** [comment]
This issue relates to what I suspect is a rare issue. The `<title>` element in EAD2002 has the full set of linking attributes, which have been removed in EAD3. For instances of `<title>` that do not have any linking attributes, the migration is as you suggest - simply migrating the title forward and adding the necessary child part.

For cases where title does have linking attributes, it's a bit more complicated. In mixed content contexts the simple solution is to wrap that title in a ref, moving the linking attributes there. That doesn't work in controlaccess because it's not mixed content and ref isn't a valid child of controlaccess. So the choice I struggled with in this scenario was to either lose the linking attributes altogether or to wrap the title in a paragraph and then a ref. I'm pretty sure it's an edge case, so it shouldn't be a problem for the vast majority of migrations.

It does occur to me that we only need to carry forward the linking attributes if there is an href value, so I'll modify the tests to only do the p/re/title tactic when href is present, rather than any of the linking attributes.
